src/sql-parser/src/parser.rs

```diff
@@ -900,7 +900,14 @@ impl<'a> Parser<'a> {
         precedence: Precedence,
     ) -> Result<Expr<Raw>, ParserError> {
         debug!("parsing infix");
-        let tok = self.next_token().unwrap(); // safe as EOF's precedence is the lowest
+        let pre_tok = self.next_token().unwrap(); // safe as EOF's precedence is the lowest
+
+        let tok = match &pre_tok {
+            Token::Keyword(OPERATOR) if self.peek_token() == Some(Token::LParen) => {
```

Comment:
I think you can use `if self.consume_token(Token::LParen)` here and then remove the expect_token at the top of parse_operator_expr.